### ISSUE TITLE
Could the icons be drawn immediately, instead of after overview animations complete?

### ISSUE THREAD

**Issue by JaneSmith** [sustmi/gnome-shell-extension-windowoverlay-icons#13]
I find this extension very useful, but it's also quite jarring for me how when I open the overlay, the windows are shown without icons as the zoom-out animation plays - and then only when everything settles into a fixed place, suddenly the icons appear.

Would it be possible to draw the icons immediately as soon as the overlay opens? So that as the zoom-out effect takes place, the icons animate along with the windows?

It would be especially great if the opacity could be animated during this transition. For example, 0% opacity on the first frame, and 100% of the last frame. So that as the windows zoom out, the icons fade in.

**Comment by sustmi** [comment]
Hello @JaneSmith .
Sorry for not replying sooner. I did not have time to check notifications in GitHub lately.

I am sorry that you do not like the current animation. To me it actually looks OK. I did not even notice it until you mentioned it.

The behavior that you propose may look better. However it seems that this would need to completely rewrite the extension.
Currently, the extension attaches to `WindowOverlay` class to display the application icons. But the windowOverlay (and also the icons) is actually displayed **after** the `zoomToOverview` animation ends.

This is the call graph from `zoomToOverview()` for the context:
- https://github.com/GNOME/gnome-shell/blob/3.36.2/js/ui/workspace.js#L1743
- https://github.com/GNOME/gnome-shell/blob/3.36.2/js/ui/workspace.js#L1286
- https://github.com/GNOME/gnome-shell/blob/3.36.2/js/ui/workspace.js#L1311
- https://github.com/GNOME/gnome-shell/blob/3.36.2/js/ui/workspace.js#L1395
- https://github.com/GNOME/gnome-shell/blob/3.36.2/js/ui/workspace.js#L1436

I am actually not sure what is the difference between (window) clone and window overlay concepts. Maybe "window overlay" only represents the border, title and close button on the overview and the window itself is represented by "window clone".
So I guess we would have to extend the `WindowClone` class (instead of `WindowOverlay`) and draw the application icons there.

However, `WindowClone` is also used in `WorkspaceThumbnail` class which represents the right bar with all desktops in the overview:
![image](https://user-images.githubusercontent.com/885946/82161547-4396fa00-988d-11ea-8258-2dbb7e7a7774.png)

Actually, I am not sure if it would be good or bad if the application icon was drawn in the WorkspaceThumbnail too.

However, this brings me to the second problem. The size of the application icon can be configured (in the extension settings) to either absolute size (in pixel) or relative size (in percents, relative to the window clone size). With my favorite setting which is absolute size of 128 pixels, the application icon would not fit inside the WorkspaceThumbnail. I would probably need to use the absolute size setting not as a "fixed size" but as "maximum size" and scale it down if the icon does not fit horizontally or vertically.

Or I could detect if the `clone.overlay` is set and only display the icons in this case (because `WorkspaceThumbnail` does not set `clone.overlay` property).

The bottom line is: it is not an easy feature to implement and the current behavior does not bother me that much. Sorry, but it I currently do not have time for this.

**Comment by elektronik2k5** [comment]
I came here to request the same feature. Thank you @sustmi for the extension and the detailed overview above. :bowing_man:

**Comment by sustmi** [comment]
Hi. I have just released new version of this extension (v36) that should fix this issue. However, it supports only GNOME Shell v3.38.
Could you, please, try the new version and let me know what you think about the transitions?

**Comment by BjoernDaase** [comment]
Highly appreciated, thanks!

I'll try it out directly after it landed on eog.

**Comment by BjoernDaase** [comment]
Tested it and it looks insane :)

**Comment by sustmi** [comment]
Thank you for your feedback @BjoernDaase . :+1:
